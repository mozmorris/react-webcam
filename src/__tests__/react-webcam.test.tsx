import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Webcam from '../react-webcam';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Webcam
        audio={false}
        audioConstraints={{
          sampleSize: 8,
          echoCancellation: true
        }}
        className="react-webcam"
        imageSmoothing={false}
        minScreenshotHeight={1000}
        minScreenshotWidth={1000}
        onUserMedia={() => {}}
        onUserMediaError={() => {}}
        screenshotFormat="image/png"
        screenshotQuality={1}
        style={{transform: 'rotate(180deg)'}}
        videoConstraints={{
          width: 160,
          height: 120,
          frameRate: 15
        }}
        height={1000}
        width={1000}
      />
    )

  expect(tree.toJSON()).toMatchSnapshot();
});

it('sets <video/> muted to false when props.audio is true', () => {
  const tree = renderer
    .create(
      <Webcam
        audio={true}
        audioConstraints={{
          sampleSize: 8,
          echoCancellation: true
        }}
        className="react-webcam"
        imageSmoothing={false}
        minScreenshotHeight={1000}
        minScreenshotWidth={1000}
        onUserMedia={() => {}}
        onUserMediaError={() => {}}
        screenshotFormat="image/png"
        screenshotQuality={1}
        style={{transform: 'rotate(180deg)'}}
        videoConstraints={{
          width: 160,
          height: 120,
          frameRate: 15
        }}
        height={1000}
        width={1000}
      />
    )

  expect(tree.root.findByType('video').props.muted).toBe(false)
})

it('sets <video/> disablePictureInPicture to true when props.disablePictureInPicture is true', () => {
  const tree = renderer
    .create(
      <Webcam
        audio={false}
        audioConstraints={{
          sampleSize: 8,
          echoCancellation: true
        }}
        className="react-webcam"
        disablePictureInPicture={true}
        imageSmoothing={false}
        minScreenshotHeight={1000}
        minScreenshotWidth={1000}
        onUserMedia={() => {}}
        onUserMediaError={() => {}}
        screenshotFormat="image/png"
        screenshotQuality={1}
        style={{transform: 'rotate(180deg)'}}
        videoConstraints={{
          width: 160,
          height: 120,
          frameRate: 15
        }}
        height={1000}
        width={1000}
      />
    )

  expect(tree.root.findByType('video').props.disablePictureInPicture).toBe(true)
})
