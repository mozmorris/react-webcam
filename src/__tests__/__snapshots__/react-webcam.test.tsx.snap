// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<video
  autoPlay={true}
  className="react-webcam"
  disablePictureInPicture={false}
  height={1000}
  muted={true}
  playsInline={true}
  style={
    Object {
      "transform": "rotate(180deg)",
    }
  }
  width={1000}
/>
`;
